devito/stencilkernel.py

```diff
@@ -513,6 +583,22 @@ def cfunction(self):
         return self._cfunction
 
     @property
+    def cfunctionJ(self):
```

Comment:
There's again some code duplication with ``cfunction``, which should be factored in a common routine

Maybe this is yet another symptom that what you want is a specialized ``StencilKernel``; that is, that you should extend ``StencilKernelBasic`` (or ``OperatorBasic``, as I called it in the new ``stencil-kernel-basic`` branch) to make things properly structured